compiler/include/baseAST.h

```diff
@@ -415,6 +427,58 @@ def_to_ast(ParamForLoop);
 
 #undef def_to_ast
 
+#define def_hash_ast(SomeType) \
+  namespace std { \
+    template<> struct less<SomeType*> { \
+      bool operator()(const SomeType* lhs, const SomeType* rhs) const { \
+        if (lhs == NULL && rhs != NULL) return true; \
+        if (lhs != NULL && rhs == NULL) return false; \
+        if (lhs == NULL && rhs == NULL) return false; \
+        return ((BaseAST*)lhs)->id < ((BaseAST*)rhs)->id; \
+      } \
+    }; \
+  }
+
+def_hash_ast(SymExpr)
```

Comment:
Oh, right, to avoid having a version we default to.  Seems painful to add one for every node if this is the behavior we always want, though